cpp/src/prims/detail/nbr_intersection.cuh

```diff
@@ -317,29 +333,42 @@ template <typename FirstElementToIdxMap,
           typename VertexPairIterator,
           typename vertex_t,
           typename edge_t,
+          typename edge_property_value_t,
+          typename edge_partition_e_input_device_view_t,
+          typename optional_property_buffer_t,
+          typename optional_idx_buffer_t,
           bool multi_gpu>
 struct copy_intersecting_nbrs_and_update_intersection_size_t {
   FirstElementToIdxMap first_element_to_idx_map{};
   raft::device_span<size_t const> first_element_offsets{};
   raft::device_span<vertex_t const> first_element_indices{nullptr};
+  raft::device_span<edge_property_value_t const> first_element_properties{nullptr};
 
   SecondElementToIdxMap second_element_to_idx_map{};
   raft::device_span<size_t const> second_element_offsets{};
   raft::device_span<vertex_t const> second_element_indices{nullptr};
+  raft::device_span<edge_property_value_t const> second_element_properties{nullptr};
 
   edge_partition_device_view_t<vertex_t, edge_t, multi_gpu> edge_partition{};
+  edge_partition_e_input_device_view_t edge_partition_e_value_input{};
 
   VertexPairIterator vertex_pair_first;
   raft::device_span<size_t const> nbr_intersection_offsets{nullptr};
   raft::device_span<vertex_t> nbr_intersection_indices{nullptr};
 
+  optional_property_buffer_t nbr_intersection_properties0{};
+  optional_property_buffer_t nbr_intersection_properties1{};
+  optional_idx_buffer_t nbr_intersection_idx_buffer{};
+
   vertex_t invalid_id{};
 
-  __device__ edge_t operator()(size_t i) const
+  __device__ edge_t operator()(size_t i)
   {
     auto pair = *(vertex_pair_first + i);
 
     vertex_t const* indices0{nullptr};
+    edge_property_value_t const* property0{nullptr};
```

Comment:
This is actually an optional span, right? (Same to nbr_intersection_properties0 and nbr_intersection_properties1). May better use the same mechanism.